include/modem/nrf_modem_lib.rst

```diff
@@ -0,0 +1,135 @@
+If your application performs an update of the nRF9160 modem firmware, you must disable this functionality to have full control on the initialization of the library.
+
+:ref:`partition_manager` is the component that reserves the RAM memory for the Modem library.
+For more information, see :ref:`partition_mgr_integration`.
+
+The library wrapper also coordinates the shutdown operation among different parts of the application that use the Modem library.
+This is done by the :c:func:`nrf_modem_lib_shutdown` function call, by invoking sleeping threads when the modem is being shut down.
+
+When using the Modem library in |NCS|, the library should be initialized and shutdown using the :c:func:`nrf_modem_lib_init` and :c:func:`nrf_modem_lib_shutdown` function calls, respectively.
+
+Socket offloading
+=================
+
+Zephyr Socket API offers the :ref:`socket offloading functionality <zephyr:net_socket_offloading>` to redirect or *offload* function calls (for example, ``socket()`` and ``send()``) to BSD socket APIs.
+The glue layer utilizes this functionality to offload the socket API calls to the Modem library and thus eases the task of porting the networking code to the nRF9160 by providing a wrapper for Modem library's native socket API.
+Examples of such wrapper functions are :c:func:`nrf_socket` and :c:func:`nrf_send`.
+
+The socket offloading functionality in the Modem library glue is implemented in :file:`nrf\\lib\\nrf_modem_lib\\nrf91_sockets.c`.
+
+Modem library socket API sets errnos as defined in :file:`nrf_errno.h`.
+The socket offloading support in the Modem library glue in |NCS| converts those errnos to the errnos that adhere to the selected C library implementation.
+
+The socket offloading functionality is enabled by default.
+To disable the functionality, set the :option:`CONFIG_NET_SOCKETS_OFFLOAD` Kconfig option to ``n`` in your project configuration.
+If you disable the socket offloading functionality, the socket calls will no longer be offloaded to the nRF9160 modem firmware.
+Instead, the calls will be relayed to the native Zephyr TCP/IP implementation.
+This can be useful to switch between an emulator and a real device while running networking code on these devices.
+Note that the even if the socket offloading is disabled, Modem library's own socket APIs such as :c:func:`nrf_socket` and :c:func:`nrf_send` remain available.
+
+OS abstraction layer
+====================
+
+For functioning, the Modem library requires the implementation of an OS abstraction layer, which is an interface over the operating system functionalities such as interrupt setup, threads, and heap.
+The Modem library glue in |NCS| provides an implementation of the Modem library OS abstraction layer using |NCS| components.
+The OS abstraction layer is implemented in the :file:`nrfxlib\\nrf_modem\\include\\nrf_modem_os.c`.
+
+The behavior of the functions in the Modem library glue is dependent on the |NCS| components that are used in their implementation.
```

Comment:
".. of the functions in the Modem library _OS abstraction layer_ .."
Should be "OS abstraction layer" and not "glue.

Reply:
Fixed